packages/next-server/server/api-utils.ts

```diff
@@ -0,0 +1,169 @@
+import { IncomingMessage } from 'http'
+import { NextApiResponse, NextApiRequest } from '../lib/utils'
+import { Stream } from 'stream'
+import getRawBody from 'raw-body'
+import { URL } from 'url'
+
+/**
+ * Parse incoming message like `json` or `urlencoded`
+ * @param req
+ */
+export async function parseBody(req: NextApiRequest) {
+  const type = req.headers['content-type'] || 'text/plain'
+  const encoding = getCharset(type)
+
+  const buffer = await getRawBody(req, { encoding })
+
+  const body = buffer.toString()
+
+  if (type.startsWith('application/json')) {
```

Comment:
Why statsWith?

Reply:
There could be encoding too `application/json; charset=utf-8` or just `application/json`

Reply:
Hmmm is this how body-parser handles it?

Reply:
Also, mime-types [are case insensitive](https://tools.ietf.org/html/rfc2045).
```
The type, subtype, and parameter names are not case sensitive.  For
example, TEXT, Text, and TeXt are all equivalent top-level media
types.
```
Probably won't happen much in practice, but this code doesn't seem to account for that. Maybe you can do:
```js
  if (type.trim().toLowerCase().startsWith('application/json')) {
```
or look for a battle-tested mime parser.

Reply:
Oh, and there's also [this](https://trac.tools.ietf.org/html/draft-ietf-appsawg-media-type-suffix-regs-02#section-3.1) as in `application/ld+json`